src/s3/commands/downloadFileAs.ts

```diff
@@ -163,3 +169,150 @@ async function promptForSaveLocation(
         filters: filters,
     })
 }
+
+async function downloadBatchFiles(fileList: S3File[], saveLocation: vscode.Uri, client: S3Client): Promise<S3File[]> {
+    const results = await Promise.all(
+        fileList.map(async file => {
+            try {
+                await downloadFile(file, {
+                    window: Window.vscode(),
+                    saveLocation: vscode.Uri.file(path.join(saveLocation.fsPath, file.name)),
+                    client,
+                    progressLocation: vscode.ProgressLocation.Notification,
+                })
+            } catch (error) {
+                getLogger().error(`s3: Failed to download file '${file.name}': `, error as Error)
+                return file
+            }
+        })
+    )
+    return results.filter(isNonNullable)
+}
+
+/**
+ * Downloads an S3 Folder or multiple selected S3 files
+ *
+ * @param node S3FolderNode to download
+ * @param allNodes Multi-selected explorer nodes
+ * @param window
+ * @param outputChannel
+ */
+export async function downloadFolderCommand(
+    node: S3FolderNode | S3FileNode,
+    allNodes: S3FileNode[] = [],
```

Comment:
The impl. doesn't really need to work with nodes.

Reply:
Having the node would be useful for multi-selected files. They can be from different buckets, but bucket info is not stored in the `S3.File`.